src/app/archive/lib/processor.ml

```diff
@@ -735,11 +765,14 @@ module Protocol_versions = struct
       (transaction, network, patch)
 
   let find_txn_version (module Conn : CONNECTION) ~transaction =
-    Conn.collect_list
-      (Caqti_request.collect Caqti_type.int Caqti_type.int
-         {sql| SELECT id FROM protocol_versions WHERE transaction = ?
-        |sql} )
-      transaction
+    let%map () = load_local_copy (module Conn) in
```

Comment:
I suggest renaming this function to something like `find_by_txn_version` (AFAIU this name more closely resembles semantics of the function)